src/metpy/plots/declarative.py

```diff
@@ -918,7 +925,13 @@ def griddata(self):
 
             if self.time is not None:
                 subset[data.metpy.time.name] = self.time
-            self._griddata = data.metpy.sel(**subset).squeeze()
+            data_subset = data.metpy.sel(**subset).squeeze()
+
+            if self.plot_units is not None:
+                data_subset.values = (data_subset.values
```

Comment:
I think you should be able to do this as `data_subset.metpy.convert_units(self.plot_units)`. That would eliminate the need to then set the unit attribute below I believe.

Reply:
I concur, as this will prevent the need for an immediate refactor after #1304.

With the decision made in #1304, I would definitely recommend that the units of a DataArray should only be handled by the accessor (since the units will soon be part of the variable data inside of the DataArray rather than an attribute on the DataArray).